src/main/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspect.kt

```diff
@@ -108,3 +116,5 @@ class OracleConnectionAspect(
 
 @Throws(SQLException::class)
 internal fun Connection.run(sql: String) = prepareStatement(sql).use { ps -> ps.execute() }
+
+class ProxyConnectionTransactionException(message: String, e: Exception) : TransactionException(message, e)
```

Comment:
throwing `TransactionException` here so that it is rethrown as is and [not wrapped](https://github.com/spring-projects/spring-framework/blob/30dc86810e2a14357bce4f2e5a6fd2faa0c0dac2/spring-orm/src/main/java/org/springframework/orm/jpa/JpaTransactionManager.java#L460-L463) in a `CannotCreateTransactionException` that is always worrying when we see it in the logs.